pants-plugins/src/python/internal_backend/sitegen/tasks/sitegen.py

```diff
@@ -251,22 +251,46 @@ def generate_site_toc(config, precomputed, here):
 
   def recurse(tree, depth_so_far):
     for node in tree:
-      if 'heading' in node:
+      if 'heading' in node and 'pages' in node:
         heading = node['heading']
+        pages = node['pages']
+        links = []
+        for i in pages:
+          if i == here:
+            link = here +'.html'
+          else:
+            link = os.path.relpath(i + '.html', os.path.dirname(here))
+          links.append(dict(link=link, 
+                            text=precomputed.page[i].title,
+                            here=(i == here)))
```

Comment:
This could probably be converted into:
```python
for cur_page in pages:
  html_filename = '{}.html'.format(cur_page)
  page_is_here = cur_page == here
  if page_is_here:
    link = html_filename
  else:
    link = os.path.relpath(html_filename, os.path.dirname(here))
  links.append(dict(
    link=link,
    text=precomputed.page[cur_page].title,
    here=page_is_here))
```

The only thing I would say should almost definitely change is just using `.format()`, which is generally preferred to string concatenation in almost all cases. The rest just seemed to make it easier to follow, especially using `'{}.html'.format(cur_page)` instead of using `here` in the first branch of the `if` -- since `i == here`, then `i + '.html'` is the same as `here + '.html'`, so it's easier to make that into a single statement. Let me know if I'm getting something wrong -- this kind of edit isn't something you need to add but I would recommend if it makes sense.